### ISSUE TITLE
Big Blind succession

### ISSUE THREAD

**Issue by philius44** [pokerth/pokerth#378]
I was left out to post the Big Blind.

Before, to my right side, the player had to post the big blind, and at the end of the round he had to leave the table. After that i just had to post the small blind, so i was left out.

**Comment by q4z1** [comment]
Hi philius44,

thx for reporting this bug. Another player also stumbled upon it.

Will be fixed in next release...

Regards.

**Comment by MarkTr** [comment]
I'm not sure about the rules, but isn't that correct behaviour? otherwise the dealer would and small blind would stay with the same Person for 2 rounds?

**Comment by doug86i** [comment]
Hi MarkTr, not quite I'm afraid.

The correct behaviour is: In a hand where the player that was the big blind the previous hand was knocked out, there will be no small blind. The big blind moves to the player it would have moved to if the other player hadn't been knocked out.

Dx

**Comment by MarkTr** [comment]
So the small will stay out of the game? Or does the dealer button stay for one round to get the small blind in again?

**Comment by albmed** [comment]
Hi, 

@MarkTr is right.  
@doug86i, there will be always small blind.  
@philius44, I think there is no bug. 
In round 24. bisestile was the dealer, Dagger was SB and TreeHunter2 was BB. 
In round 25, dealer moves one position to the left. So, Dagger was dealer. TreeHunter2 should be SB, but he left. So, he's no longer a player. Therefore, Sin42 was SB and morgana BB. 

Positions at the table are set by the player who has the button (ie, dealer). Button always moves one player to the left. SB & BB are set accordingly on dealer's left among the remaining active players (players with chips). No matter if a player leaves or is eliminated. In tournaments or in games where players can moves to other tables, this guarantees no one can be two consecutive times BB or SB. 
 
Remark: If only two players remain, dealer and SB goes to the same player.

**Comment by doug86i** [comment]
Hi All,

What I described is how the situation has been dealt with (If you'll excuse the pun) in every "real life" tournament I've played or organised.

The rule I described (where the small blind is missed on the hand after the BB is eliminated) is known as the "Dead Button" rule and as far as I have encountered, is essentially universally used in tournament play.

@MarkTr To clarify, the Small Blind is only absent for the one hand immediately after the Big Blind was eliminated. The next hand the Button moves to the "Dead" seat and the blinds are posted as normal. From then on the perviously dead seat is simply ignored.

Quoting from Wikipedia:

`Dead button: Spots vacated by leaving players who would pay the small blind or get the button during the next hand remain open for the purposes of shifting blinds and button. Thus, the small blind may not be paid in the subsequent hand if the player due to pay the small blind has vacated the spot and, therefore, is considered "dead". However, there is always a big blind even if the spot is vacated by the player who is due to pay the big blind; in such case, the player seated to the left of the vacated spot pays the big blind. When the dealer button moves to an empty seat, it also is considered "dead", and the last active player before the empty seat retains the "privilege of last action" by default.`

@albmed the rule you describe is often used in cash games where people can come and go, but I've never seen it used in tournament play. There is a variation on the rule I've seen on occasion called the 'Moving button' where the blinds move as you describe but both both blinds are then 'big' blinds (i.e both players post the same amount). I'm not keen on this rule as it seems unnecessary, in my view the Dead Blind rule is Fairer in the context of a pokerTH tournament where vacated seats are never refilled. 

Here is a link to the World Series of Poker rules from 2019.

https://www.wsop.com/2019/2019%20WSOP%20Tournament%20Rules.pdf

Point 85 states:

`Tournament play will use a dead button.`

One more link to the 2019 Poker Tournament Directors Association Rules:

https://www.dropbox.com/sh/mg0oc5e9higwvul/AACvSfnLPt73dy_Tg3azpflea?dl=0

Point 32 states that the Dead Button rule will be used.

All the best,
Dx

**Comment by albmed** [comment]
Hi @doug86i 
Thanks for your comment. I wasn't aware of this rule. 
However, as far as I know, not all tournaments use TDA rules. Some cardrooms (eg, PS, 888 and I think WPT, too) use [Robert's Rules](https://www.homepokergames.com/roberts-rules-poker.php) (at least, it's where I've been redirected when I have some doubts). 

In section 4, rule 2 (and in section 16, rule 1) it explains how button and blinds are set. It gives two options and each cardroom should choose between them: moving button or dead button. For what you said it looks WSOP uses dead button, but I think other card rooms uses moving button. Personally, I've never seen a game without a button or SB... at least, not consciously. 

Let me investigate further before making any decisions. Anyway, thanks for your explanation. 

Regards.

**Comment by MarkTr** [comment]
If at the moment moving buttons rule is used, wouldn't it make sense to add dead button rule, make it default and keep moving button as option for those who want it?

**Comment by Gitoffthelawn** [comment]
I agree with what @doug86i wrote.  Here's a simple way to remember it:

 1) Big bling ALWAYS moves to the next player.
 2) NO ONE ever skips the big blind. NO ONE ever plays it twice.
 3) Always follow rules 1 and 2.  The small blind will not exist in certain circumstances due to rules 1 and 2 above.

What I don't quite remember is if there are ANY circumstances in which a player plays the SMALL blind twice in a row.  @doug86i Do you know?

**Comment by doug86i** [comment]
@Gitoffthelawn Not as far as I'm aware.
